src/parameterizations/lateral/MOM_internal_tides.F90

```diff
@@ -423,52 +528,78 @@ subroutine propagate_int_tide(h, tv, Nb, Rho_bot, dt, G, GV, US, inttide_input_C
   ! Rotate points in the halos as necessary.
   call correct_halo_rotation(CS%En, test, G, CS%nAngle, halo=En_halo_ij_stencil)
 
+  if (CS%debug) then
+    call hchksum(CS%En(:,:,:,1,1), "EnergyIntTides af halo R", G%HI, haloshift=0, scale=US%RZ3_T3_to_W_m2*US%T_to_s)
+    do m=1,CS%nMode ; do fr=1,CS%Nfreq
+      call sum_En(G, US, CS, CS%En(:,:,:,fr,m), 'prop_int_tide: after correct halo rotation')
+      if (is_root_pe()) print *, 'prop_int_tide: after correct halo rotation', CS%En_sum
+    enddo ; enddo
+  endif
+
   ! Propagate the waves.
   do m=1,CS%nMode ; do fr=1,CS%Nfreq
 
     ! initialize residual loss, will be computed in propagate
     CS%TKE_residual_loss(:,:,:,fr,m) = 0.
+    CS%TKE_slope_loss(:,:,:,fr,m) = 0.
 
-    call propagate(CS%En(:,:,:,fr,m), cn(:,:,m), CS%frequency(fr), dt, &
-                   G, US, CS, CS%NAngle, CS%TKE_residual_loss(:,:,:,fr,m))
+    if (CS%apply_propagation) then
+      call propagate(CS%En(:,:,:,fr,m), cn(:,:,m), CS%frequency(fr), dt, &
+                     G, US, CS, CS%NAngle, CS%TKE_slope_loss(:,:,:,fr,m))
+      endif
   enddo ; enddo
 
-  ! Check for En<0 - for debugging, delete later
-  do m=1,CS%nMode ; do fr=1,CS%Nfreq ; do a=1,CS%nAngle
-    do j=js,je ; do i=is,ie
-      if (CS%En(i,j,a,fr,m)<0.0) then
-        id_g = i + G%idg_offset ; jd_g = j + G%jdg_offset
-        if (abs(CS%En(i,j,a,fr,m))>1.0) then ! only print if large
-          write(mesg,*)  'After propagation: En<0.0 at ig=', id_g, ', jg=', jd_g, &
-                         'En=', CS%En(i,j,a,fr,m)
-          call MOM_error(WARNING, "propagate_int_tide: "//trim(mesg))
-!          call MOM_error(FATAL, "propagate_int_tide: stopped due to negative energy.")
-        endif
-        CS%En(i,j,a,fr,m) = 0.0
-      endif
+  if (CS%debug) then
+    call hchksum(CS%En(:,:,:,1,1), "EnergyIntTides af prop", G%HI, haloshift=0, scale=US%RZ3_T3_to_W_m2*US%T_to_s)
+    do m=1,CS%nMode ; do fr=1,CS%Nfreq
+      call sum_En(G, US, CS, CS%En(:,:,:,fr,m), 'prop_int_tide: after propagate')
+      if (is_root_pe()) print *, 'prop_int_tide: after propagate', CS%En_sum
     enddo ; enddo
-  enddo ; enddo ; enddo
+    ! Check for En<0 - for debugging, delete later
+    do m=1,CS%nMode ; do fr=1,CS%Nfreq ; do a=1,CS%nAngle
+      do j=js,je ; do i=is,ie
+        if (CS%En(i,j,a,fr,m)<0.0) then
+          id_g = i + G%idg_offset ; jd_g = j + G%jdg_offset
+          if (abs(CS%En(i,j,a,fr,m))>1.0) then ! only print if large
```

Comment:
The 1.0 in this expression should have some units, and it is not clear to me why 1.0 should be considered a large number.

Reply:
done